packages/playwright-test/src/runner.ts

```diff
@@ -346,6 +350,8 @@ export class Runner {
       const grepInvertMatcher = project.grepInvert ? createTitleMatcher(project.grepInvert) : null;
 
       const titleMatcher = (test: TestCase) => {
+        if (doNotFilterFiles.has(test._requireFile))
```

Comment:
It seems like we apply `project.grep` to setup files when calling this function for the first time.